specs/agents/metadata.md

```diff
@@ -75,6 +76,13 @@ hostname if `configured_hostname` is not provided.
 Agents that are APM-Server-version-aware, or that are compatible only with versions >= 7.4, should
 use the new fields wherever applicable.
 
+#### Host.id
+
+APM agents MAY collect the `host.id` as an unique identifier for the host.
```

Comment:
From what I understand we have three possible options:
- the agent does not integrate with universal profiling, `host.id` is not required
- the agent integrates with universal profiling and get the `host.id` from the profiler
- the agent integrates with universal profiling and does not get the `host.id` from the profiler but should rely on the heuristic defined in otel semconv to capture it

In the 3rd case does capturing `host.id` without anything else is enough to have a host-based integration with universal profiling ?

Reply:
> In the 3rd case does capturing host.id without anything else is enough to have a host-based integration with universal profiling

I'm not sure if I understand you correctly here. The intention here is to 
 * force `host.id` to be available in the transaction documents IF the transaction documents will also contain profiler stacktraces
 * ensure that IF the agent itself already captures `host.id`, it takes precedence to avoid confusing behaviour
 
For example, the Resource Attributes we discover for `AWS` already populate the `host.id`. We want to make sure that the presence/absence of the profiling host agent doesn't affect this value, but we must print a warning if they differ.

Reply:
Thanks, I missed that the `host.id` could be provided through the resource attributes (for otel) or metadata with our APM agents.